src/read.c

```diff
@@ -3252,18 +3668,31 @@ static avifResult avifParse(avifDecoder * decoder)
                           AVIF_RESULT_BMFF_PARSE_FAILED);
             moovSeen = AVIF_TRUE;
         }
+#if defined(AVIF_ENABLE_EXPERIMENTAL_AVIR)
+        else if (!memcmp(header.type, "coni", 4)) {
+            AVIF_CHECKERR(!metaSeen && !moovSeen, AVIF_RESULT_BMFF_PARSE_FAILED);
+            AVIF_CHECKRES(avifParseCondensedImageBox(data->meta, boxOffset, boxContents.data, boxContents.size, data->diag));
+            coniSeen = AVIF_TRUE;
+        }
+
+        if (ftypSeen && !needsConi && coniSeen) {
+            // The 'coni' box should be ignored if there is no 'avir' brand, but libavif allows reading them in any order.
+            return AVIF_RESULT_NOT_IMPLEMENTED; // TODO(yguyon): Implement
+        }
+#endif // AVIF_ENABLE_EXPERIMENTAL_AVIR
 
         // See if there is enough information to consider Parse() a success and early-out:
         // * If the brand 'avif' is present, require a meta box
         // * If the brand 'avis' is present, require a moov box
-        if (ftypSeen && (!needsMeta || metaSeen) && (!needsMoov || moovSeen)) {
+        // * If the brand 'avir' is present, require a coni box
+        if (ftypSeen && (!needsMeta || metaSeen) && (!needsMoov || moovSeen) && (!needsConi || coniSeen)) {
```

Comment:
Can we put the comment "If the brand 'avir' is present, require a coni box" and the check `(!needsConi || coniSeen)` inside `#if defined(AVIF_ENABLE_EXPERIMENTAL_AVIR)`?

This may require restructuring the comments and code. Here is one possible rewrite:

```
        // See if there is enough information to consider Parse() a success and early-out:
        // ...
        // * If AVIF_ENABLE_EXPERIMENTAL_AVIR is defined and the brand 'avir' is present, require a coni box
        if (ftypSeen) {
            avifBool earlyOut = AVIF_TRUE;
            if ((needsMeta && !metaSeen) || (needsMoov && !moovSeen)) {
                earlyOut = AVIF_FALSE;
            }
#if defined(AVIF_ENABLE_EXPERIMENTAL_AVIR)
            if (needsConi && !coniSeen) {
                earlyOut = AVIF_FALSE;
            }
#endif
            if (earlyOut) {
                return AVIF_RESULT_OK;
            }
        }   
```